### PR TITLE
Adversarial Autoencoder Implementation

### PR DESCRIPTION
**Pull request by austinTalbot7241993**
This implements an adversarial autoencoder object for ancestry. No unit tests currently or validation that it works. Will do soon

### CODE REVIEW

**Comment by akotlar** [comment]
that was fast!

**Comment by austinTalbot7241993** [comment]
There are a bunch implemented online

**Comment by akotlar** [comment]
fixed some stuff in https://github.com/austinTalbot7241993/bystro/pull/8

**Review by akotlar — CHANGES_REQUESTED** [review]
See PR'd changes

File: python/python/bystro/ancestry/adversarial_autoencoder.py [hunk 1]
```diff
@@ -0,0 +1,410 @@
+"""
+Background
+----------
+Adversarial autoencoders (AAEs) represent a novel approach to
```
Comment (akotlar) [inline python/python/bystro/ancestry/adversarial_autoencoder.py:4]: Very appreciated

**Review by akotlar — APPROVED** [review]

**Comment by akotlar** [comment]
Passes our new, stricter mypy standards, and has tests demonstrating the proof of concept AAE. Future PR's will improve model performance.

**Pull request merged**